### PR TITLE
Wiki Categories Revamp & Tags Sorting

### PR DESCRIPTION
**Pull request by Yomzy2223**
# Wiki Categories Revamp & Tags Sorting

_This PR updates the Wiki Categories UI and introduces a new feature that allows users to sort wikis based on tags. This enhancement will help users find content more efficiently and provide a better browsing experience._

## How should this be tested?

1. Navigate to wiki edit page, and click the "Edit wiki details" button
2. Select any of the categories, then add some tags, and publish
3. Navigate to the category, then trying filtering with the tags

## Notes or observations

1. _On the wiki details modal, tags list is now dependent on the selected category (i.e changes with category)_
2. _Added some new tags._
3. _The filter feature has currently been commented out until editors are done adding the new tags to the wikis. Can test the filter feature here https://iq-wiki-p5tlk12w7-prediqt.vercel.app/categories/nfts/

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/3040

### CODE REVIEW

**Review by Simplypheyie96 — APPROVED** [review]

**Comment by Tolu-Mals** [comment]
I'm seeing this weird behaviour where I can select at most 4 tags, some other times at it becomes 5 tags. sometimes I can't click away from a particular tag. Then there was a time I couldn't select any tag except I clicked on 'All' first. I'll share the video soon

**Comment by Tolu-Mals** [comment]
https://app.usebubbles.com/roa3LpdTPSRDt25QdBoFBV/can-t-select-at-most-4-tags

**Comment by Tolu-Mals** [comment]
https://app.usebubbles.com/6fqsDP9ozXJWa6ZUKTPD3v/can-t-select-any-tag-till-select-all

**Comment by Tolu-Mals** [comment]
https://app.usebubbles.com/hDiW4YD44QPrFPYm7wfLQ7/can-t-deselect-first-tag

**Comment by Tolu-Mals** [comment]
https://app.usebubbles.com/5f1grqSHbn6GqsBfrhqRs3/can-t-remove-selected-tag

**Comment by Yomzy2223** [comment]
> I'm seeing this weird behaviour where I can select at most 4 tags, some other times at it becomes 5 tags. sometimes I can't click away from a particular tag. Then there was a time I couldn't select any tag except I clicked on 'All' first. I'll share the video soon

Hmm... weird! Kindly share the video, thanks.

**Comment by Tolu-Mals** [comment]
> > I'm seeing this weird behaviour where I can select at most 4 tags, some other times at it becomes 5 tags. sometimes I can't click away from a particular tag. Then there was a time I couldn't select any tag except I clicked on 'All' first. I'll share the video soon
> 
> Hmm... weird! Kindly share the video, thanks.

yeah check the links

**Comment by Yomzy2223** [comment]
> https://app.usebubbles.com/6fqsDP9ozXJWa6ZUKTPD3v/can-t-select-any-tag-till-select-all

Looks like the page had not finished loading here

**Comment by Aliiiu** [comment]
> 3\. https://iq-wiki-4zbovowel-prediqt.vercel.app/categories/nfts/

@Yomzy2223 If the max tags I can select is 4 just disabled the remaining tags or use the `cursor-not-allowed` class

**Comment by Aliiiu** [comment]
> > > https://app.usebubbles.com/roa3LpdTPSRDt25QdBoFBV/can-t-select-at-most-4-tags
> > 
> > 
> > @Yomzy2223 If the max tags I can select is 4 just disable the remaining tags or use the `cursor-not-allowed` class
> 
> True true

image
THis is nice but when I hover on it is not showing that the cursor is not allowed. Don't forget to use the class

File: src/pages/categories/[category].tsx [hunk 1]
```diff
@@ -79,6 +120,34 @@ const CategoryPage = ({
     onLoadMore: fetchMoreWikis,
   })
   const { t } = useTranslation('category')
+
+  const _handleTags = (updatedTags: string[]) => {
```
Comment (Aliiiu) [inline src/pages/categories/[category].tsx:124]: Don't forget to resolve the issue with this function

**Comment by Aliiiu** [comment]
image

How to recreate
- Go to create-wiki
- open the edit wiki modal

**Comment by Yomzy2223** [comment]
> > > > https://app.usebubbles.com/roa3LpdTPSRDt25QdBoFBV/can-t-select-at-most-4-tags
> > > 
> > > 
> > > @Yomzy2223 If the max tags I can select is 4 just disable the remaining tags or use the `cursor-not-allowed` class
> > 
> > 
> > True true
> 
> image THis is nice but when I hover on it is not showing that the cursor is not allowed. Don't forget to use the class

Yeah. Now added. Also, I think it makes sense to not disable All button when you max out, so I fixed that too. Can take a look here https://iq-wiki-p5tlk12w7-prediqt.vercel.app/categories/nfts/

**Comment by Aliiiu** [comment]
> https://iq-wiki-p5tlk12w7-prediqt.vercel.app/categories/nfts/

Nice, looks nice

File: src/pages/categories/[category].tsx (hunk 1 above)
Comment (Yomzy2223) [reply src/pages/categories/[category].tsx:124]: Was intentional. Was because the tags are currently commented out.

**Comment by Yomzy2223** [comment]
> image
> How to recreate
> 
> * Go to create-wiki
> * open the edit wiki modal

Fixed now

**Review by Simplypheyie96 — APPROVED** [review]

**Comment by Aliiiu** [comment]
> > image
> > How to recreate
> > 
> > * Go to create-wiki
> > * open the edit wiki modal
> 
> Fixed now

image

Same with the dropdown for link wikis and links

**Comment by Aliiiu** [comment]
image

@Simplypheyie96 is the dropdown suppose to be this close?

**Comment by Simplypheyie96** [comment]
> image
> @Simplypheyie96 is the dropdown supposed to be this close?

These are not issues that just arised on this pr, Except we want to fix them in this pr.

**Review by Aliiiu — APPROVED** [review]

**Comment by Aliiiu** [comment]
> > image
> > @Simplypheyie96 is the dropdown supposed to be this close?
> 
> These are not issues that just arised on this pr, Except we want to fix them in this pr.

True, i will look into it in another PR @Softdev1

**Pull request merged**